crypto/crypto/src/noise.rs

```diff
@@ -251,12 +326,16 @@ impl NoiseConfig {
         let encrypted_payload = aead
             .encrypt(nonce, msg_and_ad)
             .map_err(|_| NoiseError::Encrypt)?;
```

Comment:
We could be encrypting in place but I don't think it's worth it to optimize during the handshake. Thoughts?